website/static/js/file-browser.js

```diff
@@ -0,0 +1,284 @@
+                self.filesData(linkObject.link);
+                self.isLoadedUrl = false; // check if in fact changed
+                self.updateBreadcrumbs(linkObject);
+            }
+        };
+
+        // INFORMATION PANEL
+        self.selected = m.prop([]);
+        self.updateSelected = function(selectedList){
+            self.selected(selectedList);
+            console.log(self.selected());
+        }.bind(self);
+
+
+        // COLLECTIONS PANEL
+        self.activeCollection = m.prop(1);
+        self.updateCollection = function(coll) {
+            self.activeCollection(coll.id);
+            var linkObject = {
+                type : 'collection',
+                data : coll,
+                label : coll.label,
+                id : coll.id
+            };
+            self.updateFilesData(linkObject);
+        };
+
+
+        // USER FILTER
+        self.activeUser = m.prop(1);
+        self.updateUserFilter = function(user) {
+            self.activeUser(user.id);
+            var linkObject = {
+                type : 'user',
+                data : user,
+                label : user.label,
+                id : user.id
+            };
+            self.updateFilesData(linkObject);
+        };
+
+        // Refresh the Grid
+        self.updateList = function(element, isInit, context){
+            if(!self.isLoadedUrl) {
+                var el = element || document.getElementById('pOrganizer');
+                m.mount(el, m.component( ProjectOrganizer, { filesData : self.filesData, updateSelected : self.updateSelected, updateFilesData : self.updateFilesData}));
+                self.isLoadedUrl = true;
+            }
+        }.bind(self);
+
+        // BREADCRUMBS
+        self.updateBreadcrumbs = function(linkObject){
+            var crumb = {
+                label : linkObject.label,
+                url : linkObject.link,
+                id : linkObject.id
+            };
+            if (linkObject.type === 'collection' || linkObject.type === 'user'){
+                self.breadcrumbs([crumb]);
+                return;
+            }
+            if (linkObject.type === 'breadcrumb'){
+                self.breadcrumbs().splice(linkObject.index+1, self.breadcrumbs().length-linkObject.index-1);
+                return;
+            }
+            self.breadcrumbs().push(crumb);
+        }.bind(self);
+
+        self.generateLinks = function (linkObject) {
+            if (linkObject.type === 'collection'){
+                return $osf.apiV2Url(linkObject.data.path, linkObject.data.pathOptions);
+            }
+            if (linkObject.type === 'breadcrumb') {
+                return linkObject.data.url;
+            }
+            if (linkObject.type === 'user') {
+                return $osf.apiV2Url('users/' + linkObject.id + '/nodes', {});
+            }
+            return $osf.apiV2Url('nodes/' + linkObject.data.uid + '/children', {}); // If type === node
```

Comment:
It might be better to just do an explicit `else if`, and if you're feeling really nice add an error raising `else` branch. (Also the `if`s above could be `else if`s